spec/features/idv/doc_auth/mobile_back_image_step_spec.rb

```diff
@@ -1,79 +1,73 @@
-      expect(page).to have_content(t('doc_auth.headings.take_pic_back'))
-    end
-
-    it 'proceeds to the next page with valid info' do
-      attach_image
-      click_idv_continue
-
-      expect(page).to have_current_path(idv_doc_auth_ssn_step)
-    end
-
-    it 'allows the use of a base64 encoded data url representation of the image' do
-      unless simulate
-        assure_id = Idv::Acuant::AssureId.new
-        expect(Idv::Acuant::AssureId).to receive(:new).and_return(assure_id)
-        expect(assure_id).to receive(:post_back_image).
-          with(doc_auth_image_data_url_data).
-          and_return([true, ''])
-      end
-
-      attach_image_data_url
-      click_idv_continue
-
-      expect(page).to have_current_path(idv_doc_auth_ssn_step)
-    end
-
-    it 'proceeds to the next page if the user does not have a phone' do
-      user = create(:user, :with_authentication_app, :with_piv_or_cac)
-      sign_in_and_2fa_user(user)
-      complete_doc_auth_steps_before_mobile_back_image_step
-      attach_image
-      click_idv_continue
-
-      expect(page).to have_current_path(idv_doc_auth_ssn_step)
-    end
-
-    it 'does not proceed to the next page with invalid info' do
-      allow_any_instance_of(Idv::Acuant::AssureId).to receive(:post_back_image).
-        and_return([false, ''])
-      attach_image
-      click_idv_continue
-
-      expect(page).to have_current_path(idv_doc_auth_mobile_back_image_step) unless simulate
-    end
-
-    it 'does not proceed to the next page with result=2' do
-      allow_any_instance_of(Idv::Acuant::AssureId).to receive(:results).
-        and_return([true, assure_id_results_with_result_2])
-      attach_image
-      click_idv_continue
-
-      unless simulate
-        expect(page).to have_current_path(idv_doc_auth_mobile_front_image_step)
-        expect(page).to have_content(I18n.t('errors.doc_auth.general_error'))
-        expect(page).to have_content(strip_tags(I18n.t('errors.doc_auth.general_info'))[0..32])
-      end
-    end
+feature 'doc auth mobile back image step' do
+  include IdvStepHelper
+  include DocAuthHelper
+
+  before do
+    sign_in_and_2fa_user
+    complete_doc_auth_steps_before_mobile_back_image_step
+  end
+
+  it 'is on the correct page' do
+    expect(page).to have_current_path(idv_doc_auth_mobile_back_image_step)
+    expect(page).to have_content(t('doc_auth.headings.take_pic_back'))
+  end
+
+  it 'proceeds to the next page with valid info' do
+    attach_image
+    click_idv_continue
+
+    expect(page).to have_current_path(idv_doc_auth_ssn_step)
+  end
+
+  it 'allows the use of a base64 encoded data url representation of the image' do
+    acuant_client = AcuantMock::AcuantMockClient.new
+    expect(AcuantMock::AcuantMockClient).to receive(:new).and_return(acuant_client)
+    expect(acuant_client).to receive(:post_back_image).
+      with(hash_including(image: doc_auth_image_data_url_data)).
```

Comment:
Ditto here, could we use `mock_response!`? Is it because of the `hash_including` bit? I bet we could just update the mocks to always expect a subset of hash arguments to simplify

Reply:
This pattern appears in a few of these tests. The idea it to make sure the client is called with the correctly base64ed image.

I can try to figure out how to move this into `mock_response?`, or rethink the way we test this. Idk we want to get into argument matching in the mock client though 🤔

Reply:
We currently store the last uploaded image for the back step to process the yaml file. Maybe we could do something like that for all the images and expose it to the callers? Then we could check that the last uploaded image matches what we expect?

Reply:
Yeah the easiest thing would be just keep an array of all the params it was called with and check them at the end of the test

```ruby
expect(Acuant::AcuantMockClient.called_params(:method_name).last.image).to eq(....)
```

Reply:
I just pushed a commit that changes this up.

Reply:
https://github.com/18F/identity-idp/pull/3849/commits/49b83fa5013332e6e85c97f08aa1bd8189f07499

Reply:
love it! thank you!